scripts/build-debug-macos.sh

```
@@ -0,0 +1,54 @@
+fi
+
+echo ""
+echo "Build complete. Debug binary at:"
+echo "  ${BUILD_DIR}/src/node-exe/kth"
+echo ""
+echo "To debug: lldb ${BUILD_DIR}/src/node-exe/kth"
+
+# Run tests if specified
+if [ "$TEST" != "0" ]; then
+    echo "Running tests..."
+    cd "${BUILD_DIR}"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)